platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/attribution/AttributionParser.java

```diff
@@ -227,13 +238,23 @@ public Options withMapboxAttribution(boolean withMapboxAttribution) {
       return this;
     }
 
+    public Options withContext(Context context) {
+      this.context = new WeakReference<>(context);
+      return this;
+    }
+
     public AttributionParser build() {
       if (attributionDataStringArray == null) {
         throw new IllegalStateException("Using builder without providing attribution data");
       }
 
+      if (context == null) {
+        throw new IllegalStateException("Using builder without providing a context");
```

Comment:
Isn't this a semver-major change?

Reply:
yes, you are right, I can revisit this PR so it returns the hardcoded string in case of no context provided.

Reply:
Made this change backwards compatible, see https://github.com/mapbox/mapbox-gl-native/pull/12707/files#diff-4d31b0b92dcd5125ea5d5a966d049f7aR176 where we check for availability for context, if not available use old hardcoded string instead.